CMakeLists.txt

```diff
@@ -338,6 +338,7 @@ include_directories("src/pal/prebuilt/inc")
 if (CLR_CMAKE_PLATFORM_UNIX)
   include_directories("src/pal/inc")
   include_directories("src/pal/inc/rt")
+  include_directories("src/pal/inc/rt/cpp")
```

Comment:
If we try to include c++ headers in our code, it will end up including the ones in pal/inc/rt directory which would cause conflicts. So I moved the files that conflict with standard c++ headers to a different directory so can be optionally included if needed.